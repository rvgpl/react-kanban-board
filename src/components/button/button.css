.button {
  -webkit-appearance: none;
  background-color: var(--primaryBgColor);
  padding: 1rem;
  margin: 0;
  border: none;
  box-shadow: none;
  color: var(--primaryColor);
  font-size: 1rem;
  cursor: pointer;
}