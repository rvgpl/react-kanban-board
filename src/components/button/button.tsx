import React from "react";
import "./button.css";

type IButtonProps = {
  onClick: () => void;
  disabled?: boolean;
};
const Button: React.FC<IButtonProps> = ({
  children,
  onClick,
  disabled = false,
}) => {
  return (
    <button
      className="button"
      data-testid="button"
      onClick={onClick}
      disabled={disabled}
    >
      {children}
    </button>
  );
};

export { Button };
