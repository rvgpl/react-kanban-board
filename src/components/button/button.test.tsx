import { render, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe("Button", () => {
  it("should render", () => {
    const { container } = render(<Button onClick={jest.fn()}>Press Me</Button>);
    expect(container).toMatchSnapshot();
  });
  it("should call onclick prop on click", () => {
    const onClickMock = jest.fn();
    const { getByTestId } = render(
      <Button onClick={onClickMock}>Press Me</Button>
    );
    fireEvent.click(getByTestId("button"));
    expect(onClickMock).toBeCalledTimes(1);
  });

  it("should render a disabled button", () => {
    const onClickMock = jest.fn();
    const { getByTestId } = render(
      <Button onClick={jest.fn()} disabled>
        Press Me
      </Button>
    );

    expect(getByTestId("button")).toBeDisabled();
  });
});
