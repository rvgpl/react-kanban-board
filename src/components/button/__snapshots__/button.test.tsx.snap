// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<div>
  <button
    class="button"
    data-testid="button"
  >
    Press Me
  </button>
</div>
`;
